### ISSUE TITLE
Crash - n8n Docker - FATAL ERROR

### ISSUE THREAD

**Issue by ansred** [n8n-io/n8n#4042]
**Describe the bug**
Everytime runs a CRON workflow (Not all of them), n8n Docker container crashes

**To Reproduce**

Not sure what is causing the issue, after the latest update n8n, issue started

**Expected behavior**
To run the workflow without a crash

**Environment (please complete the following information):**

- OS: Synology DSM - Intel CPU
- n8n Version 0.193.3
- Node.js Version [e.g. 14.16.0]
- Database system [e.g. SQLite; n8n uses SQLite as default otherwise changed]
- Operation mode [e.g. own; operation modes are `own`, `main` and `queue`. Default is `own`]

**Additional context**
Error log

```
yesterday at 2:03:14 PMEditor is now accessible via:
yesterday at 2:03:14 PMhttp://localhost:5678/
today at 2:02:13 PM
today at 2:02:13 PM<--- Last few GCs --->
today at 2:02:13 PM
today at 2:02:13 PM[7:0x7feb92f2c3e0] 86367366 ms: Scavenge 2034.2 (2071.8) -> 2034.2 (2076.8) MB, 22.3 / 0.0 ms  (average mu = 0.203, current mu = 0.182) allocation failure 
today at 2:02:13 PM[7:0x7feb92f2c3e0] 86367428 ms: Scavenge 2037.0 (2076.8) -> 2037.2 (2077.8) MB, 30.3 / 0.0 ms  (average mu = 0.203, current mu = 0.182) allocation failure 
today at 2:02:13 PM[7:0x7feb92f2c3e0] 86367534 ms: Scavenge 2038.0 (2077.8) -> 2037.8 (2088.8) MB, 37.7 / 0.0 ms  (average mu = 0.203, current mu = 0.182) allocation failure 
today at 2:02:13 PM
today at 2:02:13 PM
today at 2:02:13 PM<--- JS stacktrace --->
today at 2:02:13 PM
today at 2:02:13 PMFATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
today at 2:02:33 PMInitializing n8n process
today at 2:02:46 PMn8n ready on 0.0.0.0, port 5678
today at 2:02:46 PMVersion: 0.193.3
```

**Comment by Joffcom** [comment]
Hey @ansred,

That looks like a memory issue, what version of n8n were you on before the update and what is your workflow doing? It could be unrelated to the cron itself and could just be the amount of data being worked with.

How much memory is available to n8n / docker as well?

**Comment by ansred** [comment]
@Joffcom Not sure the version exactly, but pretty sure the one before this current one I have (I keep it up to date).


I don't have any limits on the docker container, I am using the standard docker compose template via Portainer



```
############################################################
version: '3.3'
services:
  n8n:
    container_name: n8n__auto
    ports:
      - 8117:5678
    environment:
      - WEBHOOK_URL=https://n8n.domain.com/
      - EXECUTIONS_PROCESS=main
      - N8N_EMAIL_MODE=smtp
      - N8N_SMTP_HOST=smtp-relay.sendinblue.com
      - N8N_SMTP_PORT=587
      - N8N_SMTP_USER=user@domain.com
      - N8N_SMTP_PASS=xsmtpsib-5aedf2c041e58411c1097d
      - N8N_SMTP_SENDER=address@domain.com
      - N8N_SMTP_SSL=false
    volumes:
      - /volume1/docker/containers/n8n/config:/home/node/.n8n
    image: n8nio/n8n
    restart: unless-stopped

networks:
  default:
    external:
      name: Projects_Core
############################################################
```


I agree with you that it is not CRON event related, I just wanted to add that part just in case.

The worklflow does the following:

![Screen Shot 2022-09-06 at 4 04 12 PM](https://user-images.githubusercontent.com/5244872/188728140-bb147848-b9a6-4e9b-abe4-8b5cf99fe0ca.png)

Basically it SSH to a server and takes a backup and then move it to another server via SSH+SFTP

Commands used in between (SSH + SFTP + SSHPASS + ZIP + UNZIP + ls -laSh + mkdir + backup service that runs on the one servers.


Any ideas? if there is something to troubleshoot or monitor or even tweak the config on the docker?

**Comment by Joffcom** [comment]
Hey @ansred,

How much data are you dealing with in size and how much memory is available? I suspect what is happening is the NAS is running more than just n8n and might be over committed on the memory available and when trying to access what should be empty it isn't and causing it to crash.

Is it just that one workflow that fails? It could just be that it needs to be reworked to fit in with the available resources.

**Comment by ansred** [comment]
Hi @Joffcom 

So I just did the backup again workflow, n8n docker container didn't crash, so issue happens random.

The file is zipped really small in total 2MB only moving around along other files max size 50MB.


While doing the backup, I captured stats for the container in Portainer to monitor resources consumption.

Here is before and during and then after running the workflow, you can see the spike in both CPU and Memory.
![Screen Shot 2022-09-06 at 5 38 36 PM](https://user-images.githubusercontent.com/5244872/188745973-2370a6fa-aa28-4d47-8439-6c3fdfe0b84a.png)


So, is that normal to consume that much resources for a single workflow with small task + small file?

**Comment by Joffcom** [comment]
Hey @ansred,

It can be it all depends on what is actually happening, If you are running in own mode it would make a new process to run the workflow so if you have not changed that it could be having a slight impact.

n8n also keeps data in memory as it moving between nodes, The overall usage you have doesn't look that bad and I suspect the issue will come when the device is busy. It could be worth seeing if the backup workflow can be split it into sub workflows so there is less to work through but it would be interesting to leave it all running and see what the Portainer stats are like and the device itself.

**Comment by ansred** [comment]
@Joffcom makes sense, it crashed today again, the strange part that a very small size file transport worklflow takes that much memory/cpu when it runs.

I think it's a hardware limitation regardless the fact that n8n docker takes so much resources.

Thanks for the help as I don't think there is much we can do about it and we can close this issue.

**Comment by Joffcom** [comment]
Hey @ansred,

Sounds good to me, I will get this one closed for now. If it helps I know n8n runs fairly well on a Pi although I have not used it for moving data around too much. Could be an option if you have one sat in a box somewhere.
